presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -142,7 +143,7 @@ default boolean isPushdownFilterSupported(ConnectorSession session, ConnectorTab
      * This interface can be replaced with a connector optimizer rule once the engine supports these (#12546).
      */
     @Experimental
-    default ConnectorPushdownFilterResult pushdownFilter(ConnectorSession session, ConnectorTableHandle tableHandle, RowExpression filter, Optional<ConnectorTableLayoutHandle> currentLayoutHandle)
+    default ConnectorPushdownFilterResult pushdownFilter(ConnectorSession session, ConnectorTableHandle tableHandle, RowExpression filter, Optional<ConnectorTableLayoutHandle> currentLayoutHandle, ExpressionOptimizer expressionOptimizer)
```

Comment:
this change won't be necessary once you start using `rowExpressionService.getExpressionOptimizer()`